data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerClientPool.java

```diff
@@ -91,4 +113,32 @@ private WebClient getHTTPClient(final String ipAddress) {
 
         return clientBuilder.build(WebClient.class);
     }
+
+    private String getCertificateFingerPrint() {
```

Comment:
This should probably be a package-protected class specific to this purpose.

Also, the name here should be `Fingerprint` (lowercase `p`). But, this is not critical.

Reply:
I moved this functionality to the `Certificate` model class in a commit after this one for more of an OO approach. I'm open to refactoring to a helper class if you think that's a better approach?

Reply:
Yes, that is even better. Thanks!